src/pages/write-smart-contracts/billboard-tutorial.md

```diff
@@ -0,0 +1,281 @@
+`set-message` and `get-message` functions of the contract.
+
+```ts
+import { Clarinet, Tx, Chain, Account, types } from 'https://deno.land/x/clarinet@v0.10.0/index.ts';
+import { assertEquals } from 'https://deno.land/std@0.90.0/testing/asserts.ts';
+
+Clarinet.test({
+  name: 'Ensure that the message can be set',
+  async fn(chain: Chain, accounts: Map<string, Account>) {
+    let wallet_1 = accounts.get('wallet_1')!;
+
+    let assetMaps = chain.getAssetsMaps();
+    const balance = assetMaps.assets['STX'][wallet_1.address];
+
+    let block = chain.mineBlock([
+      Tx.contractCall('billboard', 'set-message', [types.utf8('testing')], wallet_1.address),
+      Tx.contractCall('billboard', 'get-message', [], wallet_1.address),
+      Tx.contractCall('billboard', 'set-message', [types.utf8('testing...')], wallet_1.address),
+      Tx.contractCall('billboard', 'get-message', [], wallet_1.address),
+    ]);
+
+    assertEquals(block.receipts.length, 4);
+    assertEquals(block.height, 2);
+
+    block.receipts[1].result.expectUtf8('testing');
+
+    block.receipts[3].result.expectUtf8('testing...');
+
+    assetMaps = chain.getAssetsMaps();
+    assertEquals(assetMaps.assets['STX'][wallet_1.address], balance - 210);
+  },
+});
+```
+
+Modify the default imports in the boilerplate to include `{ Clarinet, Tx, Chain, Account, types }` from the `clarinet`
+Deno library.
+
+Give the test a descriptive name using the `name` field, then modify the test function to include the `chain` and
+`accounts` arguments. Make sure that you define a type for each of the arguments to satisfy the Typescript compiler.
+
+Using the Clarinet library, define variables to get a wallet address principal from the Clarinet configuration, and the
+balance of that address on the chain.
+
+The functional part of the test is defined using the `chain.mineBlock()` function, which simulates the mining of a
+block. Within that function, the test makes 4 contract calls (`Tx.contractCall()`), 2 calls to `set-message` and 2 calls
+to `get-message`.
+
+Once the simulated block is mined, the test can make assertions about the chain state. This is accomplished using the
+`assertEquals()` function and the `expect` function. In this case, the test asserts that the once the simulated block
+is mined, the block height is now equal to `2`, and that the number of receipts (contract calls) in the block are
+exactly `4`.
+
+The test can then make assertions about the return values of the contract. The test checks that the result of the
+transaction calls to `get-message` match the string values that the calls to `set-message` contain. This covers the
+capability of both contract functions. Finally, the test asserts that STX were transferred from the transaction
+caller wallet, covering the price updating and token transfer.
+
+-> You have now learned how to store and update data on chain with a variable, and how to transfer STX tokens from
```

Comment:
nit: maybe we could repeat what the reader accomplished? I'm also not sure if we have a "success" note type (`->` generates an info note it seems)